contracts/src/spaces/facets/roles/IRoles.sol

```diff
@@ -48,60 +80,128 @@ interface IRolesBase {
+  /// @param roleName Name of the role
+  /// @param permissions List of permissions for the role
+  /// @param entitlements List of entitlements for the role
+  /// @return roleId Unique identifier of the created role
   function createRole(
     string calldata roleName,
     string[] memory permissions,
     CreateEntitlement[] memory entitlements
   ) external returns (uint256 roleId);
 
+  /// @notice Retrieves all roles
+  /// @return roles Array of all roles
   function getRoles() external view returns (Role[] memory roles);
 
+  /// @notice Retrieves a role by its ID
+  /// @param roleId Unique identifier of the role
+  /// @return role The role struct
   function getRoleById(uint256 roleId) external view returns (Role memory role);
 
+  /// @notice Updates an existing role
+  /// @param roleId Unique identifier of the role to update
+  /// @param roleName New name for the role
+  /// @param permissions New list of permissions for the role
+  /// @param entitlements New list of entitlements for the role
   function updateRole(
     uint256 roleId,
     string calldata roleName,
     string[] memory permissions,
     CreateEntitlement[] memory entitlements
   ) external;
 
+  /// @notice Removes a role
+  /// @param roleId Unique identifier of the role to remove
   function removeRole(uint256 roleId) external;
 
   // permissions
 
+  /// @notice Adds permissions to a role
+  /// @param roleId Unique identifier of the role
+  /// @param permissions List of permissions to add
   function addPermissionsToRole(
     uint256 roleId,
     string[] memory permissions
   ) external;
 
+  /// @notice Removes permissions from a role
+  /// @param roleId Unique identifier of the role
+  /// @param permissions List of permissions to remove
   function removePermissionsFromRole(
     uint256 roleId,
     string[] memory permissions
   ) external;
 
+  /// @notice Retrieves permissions for a role
+  /// @param roleId Unique identifier of the role
+  /// @return permissions List of permissions for the role
   function getPermissionsByRoleId(
     uint256 roleId
   ) external view returns (string[] memory permissions);
 
   // entitlements
 
+  /// @notice Adds an entitlement to a role
+  /// @param roleId Unique identifier of the role
+  /// @param entitlement Entitlement to add
   function addRoleToEntitlement(
     uint256 roleId,
     CreateEntitlement calldata entitlement
   ) external;
 
+  /// @notice Removes an entitlement from a role
+  /// @param roleId Unique identifier of the role
+  /// @param entitlement Entitlement to remove
   function removeRoleFromEntitlement(
     uint256 roleId,
     CreateEntitlement memory entitlement
   ) external;
+
+  /// @notice Sets channel permission overrides for a role
+  /// @param roleId Unique identifier of the role
+  /// @param channelId Unique identifier of the channel
+  /// @param permissions List of permissions to set as overrides
+  function setChannelPermissionOverrides(
```

Comment:
@ttillberg I needed to add these functions to the interface so they can be visible in creation of typescrip typings.